test/__snapshots__/test.js.snap

```diff
@@ -1,5 +1,75 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`with processors AsyncCallbackInterface.webidl 1`] = `
+"\\"use strict\\";
+
+const conversions = require(\\"webidl-conversions\\");
+const utils = require(\\"./utils.js\\");
+
+const interfaceName = \\"AsyncCallbackInterface\\";
+
+exports.convert = function convert(value, { context = \\"The provided value\\" } = {}) {
+  if (!utils.isObject(value)) {
+    throw new TypeError(\`\${context} is not an object.\`);
+  }
+
+  const { wrapper } = {
+    wrapper(...args) {
+      let completion;
+      let thisArg = this;
+      let O = value;
+      let X = O;
+
+      try {
+        if (typeof O !== \\"function\\") {
+          X = O[\\"asyncMethod\\"];
+          if (typeof X !== \\"function\\") {
+            throw new TypeError(\`\${context} does not correctly implement AsyncCallbackInterface.\`);
+          }
+          thisArg = O;
+        }
+
+        for (let i = 0; i < args.length; i++) {
+          args[i] = utils.tryWrapperForImpl(args[i]);
+        }
+        if (args.length < 0) {
```

Comment:
This generated code seems pretty bad, especially in the case of 0 args. In this particular case the whole thing could be omitted, including the `...args`, and you could just do `Reflect.apply(X, thisArg, [])`.

I think the guards should be unnecessary since we can always count on jsdom to call with the correct number of args. So we can just have the loop. (And we could even eliminate that, as well as the `...args` parameter declaration itself, when we know `args.length` is `0`.)

Reply:
I just copy and pasted it from <https://github.com/jsdom/webidl2js/pull/123>.

Reply:
I fixed this in my local copy.